mozci/push.py

```diff
@@ -959,6 +976,140 @@ def get_likely_regressions(self, runnable_type):
+        logger.debug(f"Got {len(groups_low)} groups with low confidence")
+
+        # Classify task groups regarding cross config failure and overall failure
+        # - if a group is failing in all tasks, then it is a "cross-config" failure
+        # - if a group is failing only in some tasks but not all, then it is not a "cross-config" failure
+        all_groups = self.group_summaries.values()
+        groups_cross_config_failure = {
+            g.name for g in all_groups if g.is_cross_config_failure
+        }
+        groups_non_cross_config_failure = {
+            g.name for g in all_groups if not g.is_cross_config_failure
+        }
+        groups_failing_in_the_push = {
+            g.name for g in all_groups if g.status == Status.FAIL
+        }
+        groups_no_confidence = {
+            g.name
+            for g in all_groups
+            if g.name not in list(bugbug_selection["groups"].keys())
+        }
+        logger.debug(
+            f"Got {len(groups_cross_config_failure)} groups with cross-config failures"
+        )
+        logger.debug(
+            f"Got {len(groups_non_cross_config_failure)} groups with no cross-config failures"
+        )
+        logger.debug(
+            f"Got {len(groups_failing_in_the_push)} groups failing in the push"
+        )
+        logger.debug(
+            f"Got {len(groups_no_confidence)} groups without bugbug confidence"
+        )
+
+        # Real failure are groups with likely regressions that were selected with high confidence
+        # AND failing across config
+        real_failures = (
+            groups_likely_regressions & groups_cross_config_failure & groups_high
+        )
+        logger.debug(f"Got {len(real_failures)} real failures")
+
+        # Intermittent failures are groups that were NOT selected (low confidence)
+        # OR without any confidence from bugbug (too low confidence)
+        # AND are not failing across config
+        intermittent_failures = groups_non_cross_config_failure & groups_low.union(
+            groups_no_confidence
+        )
+        logger.debug(f"Got {len(intermittent_failures)} intermittent failures")
+
+        # Unknown failures all the remaining failing groups that are not real nor intermittent
+        unknown_failures = (
+            groups_failing_in_the_push - real_failures - intermittent_failures
+        )
+        logger.debug(f"Got {len(unknown_failures)} unknown failures")
+
+        def _map_failing_configurations(groups):
+            # Link all the failing configurations on the given groups
+            return {
+                name: self.group_summaries[name].failing_configurations
+                for name in groups
+            }
+
+        # Output real, intermittent and unknown groupfailures
+        # along with their failing configurations
+        return Regressions(
+            real=_map_failing_configurations(real_failures),
```

Comment:
Actually, we don't want to return the configuration names, but the tasks (which is what we are going to use for the treeherder API later)

Reply:
Basically something like this is what we want to return in `real`, `intermittent` and `unknown`:
```Python
[
    task for task in self.tasks
    if any(not result.ok and result.group == self.name for result in task.results)
]
```

(where `self` is GroupSummary)

Reply:
Sounds good.